pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/BaseSegmentAssignment.java

```diff
@@ -185,7 +157,8 @@ protected Pair<List<Map<String, Map<String, String>>>, Map<String, Map<String, S
    * Rebalances segments in the current assignment using the instancePartitions and returns new assignment
    */
   protected Map<String, Map<String, String>> reassignSegments(String instancePartitionType,
-      Map<String, Map<String, String>> currentAssignment, InstancePartitions instancePartitions, boolean bootstrap) {
+      Map<String, Map<String, String>> currentAssignment, InstancePartitions instancePartitions, boolean bootstrap,
+      SegmentAssignmentStrategy segmentAssignmentStrategy, InstancePartitionsType instancePartitionsType) {
```

Comment:
Seems a bit redundant to have both `String instancePartitionType` and `InstancePartitionsType instancePartitionsType`, wondering if there's a way to collapse these (however, seems like the first `String instancePartitionType` is mixed with tier storage concept from before your change).

Reply:
Right, so kept that independent